code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -133,6 +133,16 @@ LINEN BINS
 	var/list/sheets = list()
 	var/obj/item/hidden = null
 
+/obj/structure/bedsheetbin/Destroy()
+	if(sheets)
+		for(var/sheet in sheets)
+			sheets.Remove(sheet)
```

Comment:
Don't explicitly remove the sheets from the list. Just null the list after this loop.